### PR TITLE
Provide mechanism to test build changes + meta data file + type safety

### PR DESCRIPTION
**Pull request by johnoliver**

### CODE REVIEW

**Comment by johnoliver** [comment]
run tests

File: pipelines/build/common/build_base_file.groovy [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+        context.echo "Enable tests: ${enableTests}"
+        context.echo "Publish: ${publish}"
+        context.echo "Release: ${release}"
+        context.echo "Tag/Branch name: ${scmReference}"
+
+        jobConfigurations.each { configuration ->
+            jobs[configuration.key] = {
+                IndividualBuildConfig config = configuration.value
+
+                // jdk10u-linux-x64-hotspot
```
Comment (karianna) [inline pipelines/build/common/build_base_file.groovy]: Maybe worth changing comments to 11u

File: pipelines/build/common/build_base_file.groovy [hunk 2]
```diff
@@ -0,0 +1,459 @@
...
+                        context.echo "Created job " + downstreamJobName
+                        // execute build
+                        def downstreamJob = context.build job: downstreamJobName, propagate: false, parameters: config.toBuildParams(context)
+
+                        if (downstreamJob.getResult() == 'SUCCESS') {
+                            // copy artifacts from build
+                            context.node("master") {
+                                context.catchError {
+
+                                    context.sh "rm target/${config.TARGET_OS}/${config.ARCHITECTURE}/${config.VARIANT}/* || true"
```
Comment (karianna) [inline pipelines/build/common/build_base_file.groovy]: Add comment, removing previous build

File: pipelines/build/common/build_base_file.groovy [hunk 3]
```diff
@@ -0,0 +1,459 @@
...
+                }
+            }
+        }
+        context.parallel jobs
+
+        // publish to github if needed
+        // Dont publish release automatically
+        if (publish && !release) {
+            //During testing just remove the publish
+            publishRelease()
```
Comment (karianna) [inline pipelines/build/common/build_base_file.groovy]: publishRelease --> publishBinary

File: pipelines/src/main/groovy/JenkinsTypeCheckHelperExtension.groovy [hunk 4]
```diff
@@ -0,0 +1,55 @@
+import org.codehaus.groovy.transform.stc.GroovyTypeCheckingExtensionSupport;
```
Comment (karianna) [inline pipelines/src/main/groovy/JenkinsTypeCheckHelperExtension.groovy]: delete this file

File: pipelines/src/main/groovy/BuildEnvStub.groovy [hunk 5]
```diff
@@ -0,0 +1,9 @@
+abstract class BuildEnvStub {
```
Comment (karianna) [inline pipelines/src/main/groovy/BuildEnvStub.groovy]: Worth adding a comment somewhere in this project about why you're using this approach

**Pull request merged**